tokio/src/runtime/task/list.rs

```diff
@@ -102,6 +102,26 @@ impl<S: 'static> OwnedTasks<S> {
         (join, notified)
     }
 
+    /// Bind a task that isn't safe to transfer across thread boundaries.
+    ///
+    /// # Safety
+    /// Only use this in `LocalRuntime` where the task cannot move
+    pub(crate) unsafe fn bind_local<T>(
```

Comment:
Hmm. Is it too tricky to use `LocalOwnedTasks` here?

Reply:
I figured I'd be conservative for now and just map what we do in current_thread.

We can optimize this later.